### PR TITLE
add initial set of REQ- requirements

### PR DESCRIPTION
**Pull request by danwing**

### CODE REVIEW

**Review by tireddy2 — COMMENTED** [review]
Yes, I like the proposed format for requirements.

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 1]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
...
 
-Interactive media includes content that a user can actively engage with and results in input and response actions that can be highly delay sensitive.
-They may include digital models of the real world, multimedia content and interactive engagement.
+Interactive media includes content that a user can actively engage with and results in input and response actions that can be highly delay sensitive. It also includes mixed traffic where both bulk and interactive data are exchanged within the same flow.
+They may include digital models of the real world, multimedia content, virtualized desktop/apps and interactive engagement.
 
-Examples: VoIP (peer-to-peer (P2P), group conferencing), gaming, eXtended Reality (XR).
+Examples: VoIP (peer-to-peer (P2P), group conferencing), gaming, eXtended Reality (XR), Citrix Virtual Apps and Desktops.
+
+REQ-INTERACTIVE: The importance within interactive traffic flow depends on the type of application and the user activity. Requirements for different interactive media is listed below:
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
REQ-INTERACTIVE: The importance within interactive traffic flow depends on the type of application and the user activity. Requirements for different interactive media are listed below:
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 2]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
 These are necessary to rebuild receiver state after loss of delta frames.
 The key frames are therefore more critical to deliver to the receiver than delta frames.
 
-Streaming video also contains audio frames which can be encoded separately and thus can be signaled separately.
-Audio is more critical than video for almost all applications, but its importance (relative to other packets in the flow) is still an application decision.
+Streaming video also contains audio frames which can be encoded
+separately and thus can be signaled separately (requirement:
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
separately and, thus, can be signaled separately (requirement:
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 3]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
...
+REQ-MEDIA-KEYFRAME).
+
+Audio is more critical than video for almost all applications, but its
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: I would reword this as this is not inclusive. Video or signals are important for persons with inabilities. Refer to Total Conversation concept.

Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
Audio is more critical than video for many applications, but its
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 4]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
...
+importance (relative to other packets in the flow) is still an
+application decision (requirements: REQ-MEDIA-AV-SEPARATE,
+REQ-MEDIA-CLIENT-DECIDES).
+
+Especially with media over QUIC, the server or proxy sends the same
+stream to all receivers, including the same metadata.  Thus, when a
+client needs different prioritization (e.g., video over audio), this
+is only achievable by signaling that priority inversion from the
+client to the ISP router (requirement: REQ-CLIENT-DECIDES).
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md:202]: Does this assume the host is always directly connected to the ISP network?

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 5]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
...
+REQ-MEDIA-AV-SEPARATE:  Audio can be prioritized differently than video.
 
-Examples: Super bowl, On-Demand Streaming
 
 ## Interactive Media {#uc-interactive}
 
-Interactive media includes content that a user can actively engage with and results in input and response actions that can be highly delay sensitive.
-They may include digital models of the real world, multimedia content and interactive engagement.
+Interactive media includes content that a user can actively engage with and results in input and response actions that can be highly delay sensitive. It also includes mixed traffic where both bulk and interactive data are exchanged within the same flow.
+They may include digital models of the real world, multimedia content, virtualized desktop/apps and interactive engagement.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
They may include digital models of the real world, multimedia content, virtualized desktop/apps, and interactive engagement.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 6]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
...
 
-Examples: VoIP (peer-to-peer (P2P), group conferencing), gaming, eXtended Reality (XR).
+Examples: VoIP (peer-to-peer (P2P), group conferencing), gaming, eXtended Reality (XR), Citrix Virtual Apps and Desktops.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
Examples: VoIP (Peer-to-Peer (P2P), group conferencing), gaming, eXtended Reality (XR), Virtual Apps and Desktops.
```

No need to mention a product name :-)

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 7]
```diff
@@ -152,49 +164,69 @@ Streaming video contains the occasional key frame ("I-frame") containing a full
...
+
+3. Mixed Traffic: Signal flow will vary depending on the nature of the packet. With variety of traffic going through the session, some packets can contain interactive traffic while the others contain bulk transfer. There can be combination of reliable and unreliable traffic within the same session through multiple streams. Client-to-network signaling plays a vital role in effectively forwarding mixed traffic for better user interactivity and network performance.
 
 ## User preferences {#uc-preferences}
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
## User Preferences {#uc-preferences}
```

**Review by boucadair — COMMENTED** [review]
some quick comments

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md (hunk 4 above)
Comment (danwing) [reply draft-kwbdgrr-tsvwg-net-collab-rqmts.md:202]: No, it assumes the client can tell the ISP router the priority of the incoming flow as the client wants.  The client could be behind a NAT.

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md (hunk 3 above)
Comment (danwing) [reply draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: new wording is 'many applications', and many (most) people can hear.  We are well above 80/20 threshold of the common case that audio is more critical than video, most of the time.  The ability of the receiver to change the priority by communicating to the ISP -- without cooperation of the sender -- is a requirement and gives a hearing-impaired user the ability to adjust video above audio.

Comment (boucadair) [reply draft-kwbdgrr-tsvwg-net-collab-rqmts.md]: Agreed, especially the last part of your reply.

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md (hunk 4 above)
Comment (boucadair) [reply draft-kwbdgrr-tsvwg-net-collab-rqmts.md:202]: I would zoom into that part to avoid confusion.  We can reason about "upstream router (home router, ISP router, etc.)" to cover both direct and indirect cases

**Review by boucadair — APPROVED** [review]

**Comment by boucadair** [comment]
Dan, please rebase to resolve conflicts. It is safe if you can do that check yourself. Thanks.
